Modules/Loadable/Markups/MRML/vtkMRMLMarkupsCurveNode.cxx

```diff
@@ -310,10 +310,12 @@ double vtkMRMLMarkupsCurveNode::GetCurveLength(vtkPoints* curvePoints, bool clos
     vtkGenericWarningMacro("Invalid startCurvePointIndex=" << startCurvePointIndex << ", using 0 instead");
     startCurvePointIndex = 0;
     }
-  vtkIdType lastCurvePointIndex = curvePoints->GetNumberOfPoints()-1;
-  if (numberOfCurvePoints >= 0 && startCurvePointIndex + numberOfCurvePoints - 1 < lastCurvePointIndex)
+  vtkIdType lastCurvePointIndex = curvePoints->GetNumberOfPoints() - 1;
+  bool addClosingSegment = closedCurve & startCurvePointIndex <= lastCurvePointIndex;
```

Comment:
Why do we need the `startCurvePointIndex <= lastCurvePointIndex` check?

Reply:
My thinking ... if, say, there are exactly 5 control points and the user asks for 7 points starting with number 9, we want to return 0.0 rather than the length of the closing segment.

Reply:
OK, let's take care of that explicitly, earlier - where we check for `startCurveIndex < 0`. We could add a check for `startCurveIndex >= curvePoints->GetNumberOfPoints()` for an open curve and `startCurveIndex > curvePoints->GetNumberOfPoints()` for a closed curve and return 0 immediately.